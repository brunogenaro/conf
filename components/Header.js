import React, { PureComponent } from 'react';

import Globals from '../utils/Globals';

export default class extends PureComponent {
  render() {
    return (
      <div className='header'>
        <h1 className='title'>
          React
          <span className='titleSpan'>Conf</span>
        </h1>

        <div className='hrContainer'>
          <hr className='hr'/>
        </div>

        <h5 className='subtitle'>
          {Globals.location.country} {Globals.date.year}
        </h5>

        <style jsx>{`
          .header {
            align-self: flex-start;
            justify-content: flex-start;
            margin: 30px;
            text-align: left;
          }

          .title {
            color: ${Globals.colors.white};
            margin: 0px;
            text-transform: uppercase;
            letter-spacing: 4px;
            font-weight: bold;
          }

          .titleSpan {
            color: ${Globals.colors.primary};
          }

          .hrContainer {
            width: 25px;
          }

          .hr {
            border-color: ${Globals.colors.primary};
          }

          .subtitle {
            color: ${Globals.colors.primary};
            text-transform: uppercase;
            font-size: 17px;
            margin: 0px;
          }
        `}</style>
      </div>
    );
  }
}
