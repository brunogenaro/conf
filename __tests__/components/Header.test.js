import React from 'react';
import renderer from 'react-test-renderer'
import Header from '../../components/Header.js';

describe('With Snapshot Testing', () => {
  it('Header Snapshot Testing', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot()
  })
});