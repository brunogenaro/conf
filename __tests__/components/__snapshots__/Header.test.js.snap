// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Header Snapshot Testing 1`] = `
<div
  className="header"
  data-jsx={233531283}
>
  <h1
    className="title"
    data-jsx={233531283}
  >
    React
    <span
      className="titleSpan"
      data-jsx={233531283}
    >
      Conf
    </span>
  </h1>
  <div
    className="hrContainer"
    data-jsx={233531283}
  >
    <hr
      className="hr"
      data-jsx={233531283}
    />
  </div>
  <h5
    className="subtitle"
    data-jsx={233531283}
  >
    Brasil
     
    2017
  </h5>
</div>
`;
